include/EmptySlot.h

```diff
@@ -1,8 +1,12 @@
-#pragma once
-#include "Piece.h"
-const auto EMPTY_SLOT_CHAR = '#';
-
-class EmptySlot : public Piece {
-public:
-	EmptySlot();
```

Comment:
You could consider making the EmptySlot class a singleton. Since there is only one instance of an empty slot on the board, it might make sense to enforce this constraint at the class level. This would also help ensure that the EmptySlot object is always accessed and modified in a consistent way.